### ISSUE TITLE
Make Punycode conversion optional

### ISSUE THREAD

**Issue by jurgenhaas** [voku/email-check#15]
#### What is this feature about (expected vs actual behaviour)?

Currently, the `isValid` callback always converts the local and domain part into Punycode.

However, lots of environments don't support Punycode and we'd like to validate email addresses such that special characters like German umlauts or other accented character would be recognized as invalid.

I'll be posting a PR in a minute to fix this.

#### Does it take minutes, hours or days to fix?

Minutes.

**Comment by voku** [comment]
@jurgenhaas There is a polyfill for punycode build-in (via https://github.com/symfony/polyfill-intl-idn/blob/main/Idn.php#L148) is that not working? Do you have an example?

**Comment by jurgenhaas** [comment]
@voku It's the other way round: Punycode is working but I don't want it to be involved. Because you're converting email addresses before validation with punycode, the email doesn't contain special characters anymore and therefore the validation returns the email to be correct. But in reality, the email is not correct because it contains special characters. So what we need to achieve is to not use Punycode and let the validation return with `false` so that we know it is invalid.

**Comment by voku** [comment]
@jurgenhaas Sorry, I still do not get it. We need to use punycode so that we can check that, e.g. `foobar@😍🎻😸🎩🎱🎮🍟🐝.🍕💩.ws` (https://github.com/voku/email-check/blob/master/tests/EmailCheckTest.php#L346) is a valid e-mail address.

https://3v4l.org/d50BN

**Comment by jurgenhaas** [comment]
Yes, that's correct. But in our environment, this is NOT a valid e-mail address. So, when I can turn off the Punycode feature, you will still get the same behaviour as before, but users like us can turn this off and get an error, when a user comes along with such an address.
